aioredlock/algorithm.py

```diff
@@ -67,7 +51,20 @@ def _validate_retry_delay(self, attribute, value):
     def log(self):
         return logging.getLogger(__name__)
 
-    async def lock(self, resource):
+    async def __auto_extend(self, lock):
+        """
+        Tries to reset the lock's lifetime to lock_timeout every 0.6*lock_timeout automatically
+        In case of fault the LockError exception will be raised
+        :param lock: :class:`aioredlock.Lock`
+        :raises: LockError in case of fault
+        """
+
+        await asyncio.sleep(1/3 * self.internal_lock_timeout)
+        await self.extend(lock)
+        self.watchdogs[lock.resource] = asyncio.ensure_future(self.__auto_extend(lock))
+
+
+    async def lock(self, resource, lock_timeout=-1):
```

Comment:
Add documentation for lock_timeout argument in docstring.

Reply:
cool